java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/GenericMqttFactory.java

```diff
@@ -0,0 +1,23 @@
+package com.cumulocity.generic.mqtt.client;
+
+/**
+ * {@link GenericMqttFactory} is used to configure and create instances of {@link GenericMqttPublisher} and {@link GenericMqttSubscriber}.
+ */
+public interface GenericMqttFactory {
+
+    /**
+     * Creates an instance of <code>GenericMqttPublisher</code> with the configured topic.
+     *
+     * @param config <code>GenericMqttConnectionConfig</code>
```

Comment:
praise: I like the convention of only linking once and using code tags thereafter.